test-vectors/tx-out-records/build.rs

```diff
@@ -0,0 +1,161 @@
+use mc_account_keys::AccountKey;
+use mc_common::ResponderId;
+use mc_crypto_keys::RistrettoPrivate;
+use mc_fog_ingest_enclave_api::{IngestEnclave, IngestEnclaveInitParams};
+use mc_fog_ingest_enclave_impl::SgxIngestEnclave;
+use mc_fog_types::{
+    ingest::TxsForIngest,
+    view::{FogTxOut, TxOutRecord},
+};
+use mc_fog_view_protocol::UserPrivate;
+use mc_oblivious_traits::HeapORAMStorageCreator;
+use mc_test_vectors_definitions::tx_out_records::{
+    CorrectTxOutRecordData, IncorrectTxOutRecordData,
+};
+use mc_transaction_core::{fog_hint::FogHint, tokens::Mob, tx::TxOut, Amount, Token};
+use mc_util_from_random::FromRandom;
+use mc_util_test_vector::write_jsonl;
+use rand::{rngs::StdRng, SeedableRng};
+
+fn main() {
+    write_correct_tx_out_records();
+    write_incorrect_tx_out_records();
+}
+
+fn write_correct_tx_out_records() {
+    write_jsonl("../vectors", || {
+        let mut correct_tx_out_record_data_set: Vec<CorrectTxOutRecordData> = Vec::new();
+        let tx_out_record_data = generate_tx_out_record_data();
+        for tx_out_record in &tx_out_record_data.tx_out_records {
+            let correct_tx_out_record_data = CorrectTxOutRecordData {
+                recipient_view_private_key_hex_proto_bytes: hex::encode(mc_util_serial::encode(
+                    &tx_out_record_data.recipient_view_private_key,
+                )),
+                tx_out_record_hex_proto_bytes: hex::encode(mc_util_serial::encode(tx_out_record)),
+            };
+
+            correct_tx_out_record_data_set.push(correct_tx_out_record_data);
+        }
+
+        correct_tx_out_record_data_set
```

Comment:
Optional/Equivalent: Use iterator chaining, e.g.
```suggestion
        let tx_out_record_data = generate_tx_out_record_data();
        let private_key_hex_proto_bytes = hex::encode(mc_util_serial::encode(
            &tx_out_record_data.recipient_view_private_key,
        ));
        tx_out_record_data
            .tx_out_records 
            .iter()
            .map(|tx_out_record| {
                CorrectTxOutRecordData {
                    recipient_view_private_key_hex_proto_bytes: private_key_hex_proto_bytes.clone(),
                    tx_out_record_hex_proto_bytes: hex::encode(mc_util_serial::encode(tx_out_record)),
                }
            })
            .collect()
```

Reply:
I think this is a good suggestion, and I will implement this in a follow up PR.